File: Cpp/1833_Maximum_Ice_Cream_Bars.cpp
class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        int result = 0;
        sort(costs.begin(), costs.end());
        
        int total_price = 0;
        for(auto& cost : costs) {
            total_price += cost;
            if(total_price > coins) return result;
            
            ++result;
        }
        
        return result;
    }
};

File: Kotlin/1833_Maximum_Ice_Cream_Bars.kt
class Solution {
    fun maxIceCream(costs: IntArray, coins: Int): Int {
        var result = 0
        costs.sort()
        
        var totalPrice = 0
        for(cost in costs) {
            totalPrice += cost
            if(totalPrice > coins) return result
            
            ++result
        }
        
        return result
    }
}